controllers/component/component_adapter.go

```diff
@@ -105,12 +106,17 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 			return controller.RequeueWithError(err)
 		}
 	}
-	err = h.RemoveFinalizerFromComponent(a.client, a.logger, a.context, a.component, h.ComponentFinalizer)
-	if err != nil {
-		return controller.RequeueWithError(fmt.Errorf("failed to remove the finalizer: %w", err))
+	a.component, err = a.loader.GetComponent(a.client, a.context, a.component.Name, a.component.Namespace)
+	if errors.IsNotFound(err) {
```

Comment:
I don't understand this. if component doesn't exist, how can we remove finalizer?

Reply:
Should not this be other way around, if error Is**Not**Found, meaning we could get component and we can remove finalizer in that case.

Reply:
I have probably misundersoot using errors.IsNotFound(err)
The way I thaught of it:
1. we try to get component
2. if there is no error > remove finalizer
3. in case of error getting component > inform about such fact and requeue with error
4. In case of error removing the finalizer do the same as in 3.

Reply:
I changed the logic, let me know if its correct this itme.